### ISSUE TITLE
When modal opened , click the scroll will make any control lose focus.

### ISSUE THREAD

**Issue by heroboy** [react-bootstrap/react-bootstrap#5518]
## To Reproduce

1. goto the example page.
2. click the inputbox on page.
3. click the scrollbar, the inputbox doesn't lose focus.
4. click show modal button.
5. click the inputbox in the modal.
6. click the scrollbar, the inpubox lose the focus.

## Reproducible Example

https://codesandbox.io/s/hidden-cherry-j9ck1?file=/src/App.js

## Expected behavior
The input box not lose the focus in step 6.

## Environment (please complete the following information)

- Operating System: window
- Browser, Version : firefox 82, chrome 86
- React-Bootstrap Version : see the example

**Comment by kyletsang** [comment]
I noticed this in upstream bootstrap and other modals from other libraries as well.  Other than the inconsistent behavior, is it actually expected for inputs to retain focus when you click away on the scrollbar?

**Comment by heroboy** [comment]
>  Other than the inconsistent behavior, is it actually expected for inputs to retain focus when you click away on the scrollbar?

Yes. I'm making a dropdown with input box like [this example](https://react-bootstrap.github.io/components/dropdowns/#custom-dropdown-components). The dropdown is long, and with a scrollbar. If the dropdown is in a modal, click the scrollbar in the dropdown will make the filter input box lose focus.

**Comment by jquense** [comment]
there isn't much we can do about this unfortunately. Browsers don't provide the necessary API's for natively limiting focus for custom modals (the `<dialog>` element  is not well supported). So we have to do some hacky things to move it around when it leaves the area. This disrupts some of the browsers normal heuristics for managing focus but is still better than doing nothing.

**Comment by heroboy** [comment]
@jquense , I guess you have do something that make it losing focus. Because I test many ways that clicking the scrollbar will not make input lose focus. Can you give me some simple example using plain html and js, that when clicking the scrollbar will make input losing focus?

**Comment by jquense** [comment]
i don't have such an example, my point was that we **are** messing with the focus in Modals. Modals require that focus not drop to the underlying page, to do this we need to listen to blur and focus events and when we see that focus blurs outside the modal we forceable move it back to the first focusable element in the modal (the dialog element itself). I haven't tested this specific case but what i think is happening is that focus _does_  drop when you click the scrollbar (or at least onblur fires), however Chrome (and others) helpfully put the focus right back if nothing happens inbetween. In the modal case tho, something does happen, we see that focus as moved out of the modal, and forceable move it back.

**Comment by heroboy** [comment]
I update the example: https://codesandbox.io/s/interesting-rain-1b6g4
I add a scrollbar in the page element.
I add a scrollbar in dropdown.
I add a [typeahead](https://github.com/ericgio/react-bootstrap-typeahead).(It is strange that clicking typeahead's scrollbar won't make input lose focus.)
I tried add `enforceFocus={false}`, `restoreFocus={false}`.


And I tried using browser developer tools to remove all events on the page as possible, still nothing happened.
